packages/terra-navigation-prompt/src/NavigationPromptCheckpoint.jsx

```diff
@@ -0,0 +1,148 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import PromptRegistrationContext from './PromptRegistrationContext';
+import withPromptRegistration from './_withPromptRegistration';
+import CheckpointNotificationDialog from './_CheckpointNotificationDialog';
+
+const propTypes = {
+  /**
+   * Components to render within the context of the NavigationPromptCheckpoint. Any NavigationPrompts rendered within
+   * these child components will be registered to this NavigationPromptCheckpoint instance.
+   */
+  children: PropTypes.node,
+  /**
+   * A function that will be executed when NavigationPrompts are registered to or deregistered from the NavigationPromptCheckpoint instance.
+   * This can be used to track registered prompts outside of a NavigationPromptCheckpoint and handle prompt resolution directly, if necessary.
+   * The function will be provided with an array of data objects representing the registered NavigationPrompts as the sole argument. An empty
+   * Array will be provided when no prompts are registered.
+   *
+   * Function Example:
+   *
+   * `(arrayOfPrompts) => {
+   *   arrayOfPrompts.forEach((prompt) => {
+   *     console.log(prompt.description);
+   *     console.log(prompt.metaData);
+   *   })
+   *   this.myLocalPromptRegistry = arrayOfPrompts;
+   * }`
+   */
+  onPromptChange: PropTypes.func,
+  /**
+   * @private
+   * An object containing prompt registration APIs provided through the PromptRegistrationContext.
+   */
+  promptRegistration: PropTypes.shape({
+    registerPrompt: PropTypes.func.isRequired,
```

Comment:
could `isDefaultContextValue: PropTypes.bool,` be passed?

Reply:
It will be passed with the default context value. This will happen for the top-most checkpoint in the hierarchy. I didn't include it here because the NavigationPromptCheckpoint doesn't actually use that value and thus doesn't really have an expectation that it is there, even optionally. But I think I'll export the shape from the context and reuse it here and in the NavigationPrompt, rendering this all moot.

Reply:
68704dd